src/UIAutomation/UiaOperationAbstraction/UiaOperationAbstraction.cpp

```diff
@@ -2117,7 +2126,60 @@ namespace UiaOperationAbstraction
         return AsType<UiaString>();
     }
 
-    UiaVariant::operator std::shared_ptr<wil::unique_variant>() const
+    UiaBool UiaVariant::IsElement() const
+    {
+        if (ShouldUseRemoteApi())
+        {
+            auto remoteObject = std::get<RemoteType>(m_member);
+            auto remoteAny = remoteObject.try_as<AutomationRemoteAnyObject>();
+            if (remoteAny)
+            {
+                return remoteAny.IsElement();
+            }
+            auto remoteDerived = remoteObject.try_as<typename UiaElement::RemoteType>();
+                    if (remoteDerived)
+                    {
+                        return true;
+                    }
+            return false;
+        }
+        auto localValue = std::get<typename LocalType>(m_member);
+        if(V_VT(localValue) != UiaElement::c_comVariantType) {
+            return false;
+        }
+        winrt::com_ptr<IUnknown> punk;
+        punk.copy_from(localValue->punkVal);
+        if(punk.try_as<UiaElement::LocalType::type>()) {
+            return true;
+        }
+        return false;
+    }
+
+    UiaElement UiaVariant::AsElement() const

```

Comment:
I need to think about this some more, but is it possible to use a similar approach here to my comment on `IsElement`, where we first call into `AsType<UiaElement>` and special case the result for local types? Or does that fail to compile?

Reply:
I think doing that with HasElement fails to compile. I can't quite remember the error right now. However, I will play around with it some more.

Reply:
Ah yes, because the return type would be wrong. You can't convert an IUnknown* to a UiaElement.